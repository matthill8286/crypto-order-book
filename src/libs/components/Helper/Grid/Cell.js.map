{"version":3,"sources":["../../../../src/components/Helper/Grid/Cell.tsx"],"names":["React","StyledCell","Cell","tag","rest","Offset","offsetProps","columns","colsXs","colsSm","colsMd","colsLg","colsXl"],"mappings":";;;;;;;;;;;;AAAA,OAAOA,KAAP,MAAyC,OAAzC;AACA,SAASC,UAAT;AAGA,OAAO,IAAMC,IAAkC,GAAG,SAArCA,IAAqC;AAAA,sBAAGC,GAAH;AAAA,MAAGA,GAAH,yBAAS,KAAT;AAAA,MAAmBC,IAAnB;;AAAA,SAChD,oBAAC,UAAD;AAAY,IAAA,EAAE,EAAED;AAAhB,KAAyBC,IAAzB,EADgD;AAAA,CAA3C;AAIP,OAAO,IAAMC,MAAsC,GAAG,SAAzCA,MAAyC,QAA8B;AAAA,wBAA3BF,GAA2B;AAAA,MAA3BA,GAA2B,0BAArB,KAAqB;AAAA,MAAXC,IAAW;;AAClF,MAAIE,WAAJ;;AAGA,MAAIF,IAAI,CAACG,OAAT,EAAkB;AAChBD,IAAAA,WAAW,GAAGF,IAAd;AACD,GAFD,MAEO;AACLE,IAAAA,WAAW,mCACNF,IADM;AAETI,MAAAA,MAAM,EAAEJ,IAAI,CAACI,MAAL,IAAe,CAFd;AAGTC,MAAAA,MAAM,EAAEL,IAAI,CAACK,MAAL,IAAe,CAHd;AAITC,MAAAA,MAAM,EAAEN,IAAI,CAACM,MAAL,IAAe,CAJd;AAKTC,MAAAA,MAAM,EAAEP,IAAI,CAACO,MAAL,IAAe,CALd;AAMTC,MAAAA,MAAM,EAAER,IAAI,CAACQ,MAAL,IAAe;AANd,MAAX;AAQD;;AACD,SAAO,oBAAC,UAAD;AAAY,IAAA,EAAE,EAAET;AAAhB,KAAyBG,WAAzB,EAAP;AACD,CAjBM","sourcesContent":["import React, { FunctionComponent } from 'react'\nimport { StyledCell } from './Cell.styled'\nimport { CellProps, OffsetProps } from './Grid.interface'\n\nexport const Cell: FunctionComponent<CellProps> = ({ tag = 'div', ...rest }) => (\n  <StyledCell as={tag} {...rest} />\n)\n\nexport const Offset: FunctionComponent<OffsetProps> = ({ tag = 'div', ...rest }) => {\n  let offsetProps: Omit<OffsetProps, 'tag' | 'children'>\n\n  // According to the unit tests the Offset component defaulted to display: none if no props are given\n  if (rest.columns) {\n    offsetProps = rest\n  } else {\n    offsetProps = {\n      ...rest,\n      colsXs: rest.colsXs || 0,\n      colsSm: rest.colsSm || 0,\n      colsMd: rest.colsMd || 0,\n      colsLg: rest.colsLg || 0,\n      colsXl: rest.colsXl || 0,\n    }\n  }\n  return <StyledCell as={tag} {...offsetProps} />\n}\n"],"file":"Cell.js"}