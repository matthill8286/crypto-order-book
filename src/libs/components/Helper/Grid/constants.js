import { saiyanTheme } from "../../../styles/sc-vars-saiyan";
var columnWidth = 9 * saiyanTheme.defaultSpacing;
var gutterWidth = {
  xs: 2 * saiyanTheme.defaultSpacing,
  sm: 2 * saiyanTheme.defaultSpacing,
  md: 3 * saiyanTheme.defaultSpacing,
  lg: 3 * saiyanTheme.defaultSpacing,
  xl: 3 * saiyanTheme.defaultSpacing
};
var gridMargins = {
  xs: 2 * saiyanTheme.defaultSpacing,
  sm: 3 * saiyanTheme.defaultSpacing,
  md: 4 * saiyanTheme.defaultSpacing,
  lg: 6 * saiyanTheme.defaultSpacing,
  xl: 8 * saiyanTheme.defaultSpacing
};
var fixedColumnValues = {
  xs: {
    totalWidth: 4 * columnWidth + 3 * gutterWidth.xs + 2 * gridMargins.xs
  },
  sm: {
    totalWidth: 8 * columnWidth + 7 * gutterWidth.sm + 2 * gridMargins.sm
  },
  md: {
    totalWidth: 8 * columnWidth + 7 * gutterWidth.md + 2 * gridMargins.md
  },
  lg: {
    totalWidth: 12 * columnWidth + 11 * gutterWidth.lg + 2 * gridMargins.lg
  },
  xl: {
    totalWidth: 12 * columnWidth + 11 * gutterWidth.xl + 2 * gridMargins.xl
  }
};
export { fixedColumnValues, gridMargins, gutterWidth };
//# sourceMappingURL=constants.js.map