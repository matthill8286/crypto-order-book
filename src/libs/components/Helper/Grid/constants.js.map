{"version":3,"sources":["../../../../src/components/Helper/Grid/constants.ts"],"names":["saiyanTheme","columnWidth","defaultSpacing","gutterWidth","xs","sm","md","lg","xl","gridMargins","fixedColumnValues","totalWidth"],"mappings":"AAAA,SAASA,WAAT;AAEA,IAAMC,WAAW,GAAG,IAAID,WAAW,CAACE,cAApC;AACA,IAAMC,WAAW,GAAG;AAClBC,EAAAA,EAAE,EAAE,IAAIJ,WAAW,CAACE,cADF;AAElBG,EAAAA,EAAE,EAAE,IAAIL,WAAW,CAACE,cAFF;AAGlBI,EAAAA,EAAE,EAAE,IAAIN,WAAW,CAACE,cAHF;AAIlBK,EAAAA,EAAE,EAAE,IAAIP,WAAW,CAACE,cAJF;AAKlBM,EAAAA,EAAE,EAAE,IAAIR,WAAW,CAACE;AALF,CAApB;AAQA,IAAMO,WAAW,GAAG;AAClBL,EAAAA,EAAE,EAAE,IAAIJ,WAAW,CAACE,cADF;AAElBG,EAAAA,EAAE,EAAE,IAAIL,WAAW,CAACE,cAFF;AAGlBI,EAAAA,EAAE,EAAE,IAAIN,WAAW,CAACE,cAHF;AAIlBK,EAAAA,EAAE,EAAE,IAAIP,WAAW,CAACE,cAJF;AAKlBM,EAAAA,EAAE,EAAE,IAAIR,WAAW,CAACE;AALF,CAApB;AAQA,IAAMQ,iBAAiB,GAAG;AACxBN,EAAAA,EAAE,EAAE;AACFO,IAAAA,UAAU,EAAE,IAAIV,WAAJ,GAAkB,IAAIE,WAAW,CAACC,EAAlC,GAAuC,IAAIK,WAAW,CAACL;AADjE,GADoB;AAIxBC,EAAAA,EAAE,EAAE;AACFM,IAAAA,UAAU,EAAE,IAAIV,WAAJ,GAAkB,IAAIE,WAAW,CAACE,EAAlC,GAAuC,IAAII,WAAW,CAACJ;AADjE,GAJoB;AAOxBC,EAAAA,EAAE,EAAE;AACFK,IAAAA,UAAU,EAAE,IAAIV,WAAJ,GAAkB,IAAIE,WAAW,CAACG,EAAlC,GAAuC,IAAIG,WAAW,CAACH;AADjE,GAPoB;AAUxBC,EAAAA,EAAE,EAAE;AACFI,IAAAA,UAAU,EAAE,KAAKV,WAAL,GAAmB,KAAKE,WAAW,CAACI,EAApC,GAAyC,IAAIE,WAAW,CAACF;AADnE,GAVoB;AAaxBC,EAAAA,EAAE,EAAE;AACFG,IAAAA,UAAU,EAAE,KAAKV,WAAL,GAAmB,KAAKE,WAAW,CAACK,EAApC,GAAyC,IAAIC,WAAW,CAACD;AADnE;AAboB,CAA1B;AAkBA,SAASE,iBAAT,EAA4BD,WAA5B,EAAyCN,WAAzC","sourcesContent":["import { saiyanTheme } from '@/styles/sc-vars-saiyan'\n\nconst columnWidth = 9 * saiyanTheme.defaultSpacing\nconst gutterWidth = {\n  xs: 2 * saiyanTheme.defaultSpacing,\n  sm: 2 * saiyanTheme.defaultSpacing,\n  md: 3 * saiyanTheme.defaultSpacing,\n  lg: 3 * saiyanTheme.defaultSpacing,\n  xl: 3 * saiyanTheme.defaultSpacing,\n}\n\nconst gridMargins = {\n  xs: 2 * saiyanTheme.defaultSpacing,\n  sm: 3 * saiyanTheme.defaultSpacing,\n  md: 4 * saiyanTheme.defaultSpacing,\n  lg: 6 * saiyanTheme.defaultSpacing,\n  xl: 8 * saiyanTheme.defaultSpacing,\n}\n\nconst fixedColumnValues = {\n  xs: {\n    totalWidth: 4 * columnWidth + 3 * gutterWidth.xs + 2 * gridMargins.xs,\n  },\n  sm: {\n    totalWidth: 8 * columnWidth + 7 * gutterWidth.sm + 2 * gridMargins.sm,\n  },\n  md: {\n    totalWidth: 8 * columnWidth + 7 * gutterWidth.md + 2 * gridMargins.md,\n  },\n  lg: {\n    totalWidth: 12 * columnWidth + 11 * gutterWidth.lg + 2 * gridMargins.lg,\n  },\n  xl: {\n    totalWidth: 12 * columnWidth + 11 * gutterWidth.xl + 2 * gridMargins.xl,\n  },\n}\n\nexport { fixedColumnValues, gridMargins, gutterWidth }\n"],"file":"constants.js"}