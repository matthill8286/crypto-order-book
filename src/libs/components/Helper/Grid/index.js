export * from "./Cell";
export * from "./Grid";
export * from "./Row";
export * from "./Grid.interface";
//# sourceMappingURL=index.js.map