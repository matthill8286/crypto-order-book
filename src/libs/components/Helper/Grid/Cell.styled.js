import { media } from "../../../styles/media";
import { css, styled } from "../../../styles/styled";
import { gutterWidth } from "./constants";

var getXsCols = function getXsCols(_ref) {
  var columns = _ref.columns,
      colsXs = _ref.colsXs;
  var cols = colsXs !== undefined ? colsXs : columns;

  if (cols !== undefined) {
    switch (cols) {
      case 0:
        return "\n          display: none;\n        ";

      case 1:
        return css(["width:calc(25% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 1;}"], gutterWidth.xs);

      case 2:
        return css(["width:calc(50% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 2;}"], gutterWidth.xs);

      case 3:
        return css(["width:calc(75% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 3;}"], gutterWidth.xs);

      default:
        return css(["width:calc(100% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 4;}"], gutterWidth.xs);
    }
  }
};

var getSmMdCols = function getSmMdCols(md) {
  return function (_ref2) {
    var columns = _ref2.columns,
        colsSm = _ref2.colsSm,
        colsMd = _ref2.colsMd;
    var cols;

    if (md) {
      cols = colsMd !== undefined ? colsMd : columns;
    } else {
      cols = colsSm !== undefined ? colsSm : columns;
    }

    var gutter = md ? gutterWidth.md : gutterWidth.sm;

    if (cols !== undefined) {
      switch (cols) {
        case 0:
          return "\n          display: none;\n        ";

        case 1:
          return css(["width:calc(12.5% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 1;}"], gutter);

        case 2:
          return css(["width:calc(25% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 2;}"], gutter);

        case 3:
          return css(["width:calc(37.5% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 3;}"], gutter);

        case 4:
          return css(["width:calc(50% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 4;}"], gutter);

        case 5:
          return css(["width:calc(62.5% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 5;}"], gutter);

        case 6:
          return css(["width:calc(75% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 6;}"], gutter);

        case 7:
          return css(["width:calc(87.5% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 7;}"], gutter);

        default:
          return css(["width:calc(100% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 8;}"], gutter);
      }
    }
  };
};

var getLgXlCols = function getLgXlCols(xl) {
  return function (_ref3) {
    var columns = _ref3.columns,
        colsLg = _ref3.colsLg,
        colsXl = _ref3.colsXl;
    var cols;

    if (xl) {
      cols = colsXl !== undefined ? colsXl : columns;
    } else {
      cols = colsLg !== undefined ? colsLg : columns;
    }

    var gutter = xl ? gutterWidth.xl : gutterWidth.lg;

    if (cols !== undefined) {
      switch (cols) {
        case 0:
          return "\n          display: none;\n        ";

        case 1:
          return css(["width:calc(8.33333% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 1;}"], gutter);

        case 2:
          return css(["width:calc(16.66667% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 2;}"], gutter);

        case 3:
          return css(["width:calc(25% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 3;}"], gutter);

        case 4:
          return css(["width:calc(33.33333% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 4;}"], gutter);

        case 5:
          return css(["width:calc(41.66667% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 5;}"], gutter);

        case 6:
          return css(["width:calc(50% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 6;}"], gutter);

        case 7:
          return css(["width:calc(58.33333% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 7;}"], gutter);

        case 8:
          return css(["width:calc(66.66667% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 8;}"], gutter);

        case 9:
          return css(["width:calc(75% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 9;}"], gutter);

        case 10:
          return css(["width:calc(83.33333% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 10;}"], gutter);

        case 11:
          return css(["width:calc(91.66667% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 11;}"], gutter);

        default:
          return css(["width:calc(100% - ", "px);@supports (display:grid){width:auto;grid-column-end:span 12;}"], gutter);
      }
    }
  };
};

export var StyledCell = styled.div.withConfig({
  displayName: "Cellstyled__StyledCell",
  componentId: "sc-4z19ev-0"
})(["flex-grow:1;box-sizing:border-box;margin:calc(", "px / 2);width:calc(100% - ", "px);@supports (display:grid){margin:0;width:auto;grid-column-end:span 4;}", " ", "{display:", ";margin:calc(", "px / 2);width:calc(50% - ", "px);@supports (display:grid){margin:0;width:auto;grid-column-end:span 4;}", "}", "{display:", ";margin:calc(", "px / 2);width:calc(50% - ", "px);@supports (display:grid){margin:0;width:auto;grid-column-end:span 4;}", "}", "{display:", ";margin:calc(", "px / 2);width:calc(33.33333% - ", "px);@supports (display:grid){margin:0;width:auto;grid-column-end:span 4;}", "}", "{display:", ";margin:calc(", "px / 2);width:calc(33.33333% - ", "px);@supports (display:grid){margin:0;width:auto;grid-column-end:span 4;}", "}", "{", "}", " ", " ", " ", ""], gutterWidth.xs, gutterWidth.xs, getXsCols, media.sm, function (_ref4) {
  var alignHeight = _ref4.alignHeight;
  return alignHeight ? "flex" : "block";
}, gutterWidth.sm, gutterWidth.sm, getSmMdCols(false), media.md, function (_ref5) {
  var alignHeight = _ref5.alignHeight;
  return alignHeight ? "flex" : "block";
}, gutterWidth.md, gutterWidth.md, getSmMdCols(true), media.lg, function (_ref6) {
  var alignHeight = _ref6.alignHeight;
  return alignHeight ? "flex" : "block";
}, gutterWidth.lg, gutterWidth.lg, getLgXlCols(false), media.xl, function (_ref7) {
  var alignHeight = _ref7.alignHeight;
  return alignHeight ? "flex" : "block";
}, gutterWidth.xl, gutterWidth.xl, getLgXlCols(true), media.ie11, function (_ref8) {
  var noMargin = _ref8.noMargin;
  return noMargin && "\n        margin: 0;\n    ";
}, function (_ref9) {
  var order = _ref9.order;
  return order && css(["order:", ";"], order);
}, function (_ref10) {
  var align = _ref10.align;

  if (align) {
    switch (align) {
      case 'bottom':
        return "\n            align-self: flex-end;\n\n            @supports (display: grid) {\n              align-self: end;\n            }\n            ";

      case 'middle':
        return "\n            align-self: center;\n          ";

      case 'top':
        return "\n          align-self: flex-start;\n\n            @supports (display: grid) {\n              align-self: start;\n            }\n          ";

      default:
        return '';
    }
  }
}, function (_ref11) {
  var justify = _ref11.justify;

  if (justify) {
    switch (justify) {
      case 'flex-end':
        return "\n            justify-self: flex-end;\n\n            @supports (display: grid) {\n              justify-self: end;\n            }\n            ";

      case 'stretch':
        return "\n            justify-self: stretch;\n          ";

      case 'center':
        return "\n            justify-self: center;\n          ";

      case 'flex-start':
        return "\n          justify-self: flex-start;\n\n            @supports (display: grid) {\n              justify-self: start;\n            }\n          ";

      default:
        return '';
    }
  }
}, function (_ref12) {
  var alignHeight = _ref12.alignHeight;
  return alignHeight && "display: flex;\n      > div {\n        flex: 1\n      }";
});
//# sourceMappingURL=Cell.styled.js.map