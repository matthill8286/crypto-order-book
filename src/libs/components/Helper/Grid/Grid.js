function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { media } from "../../../styles/media";
import { css, styled } from "../../../styles/styled";
import { fixedColumnValues, gridMargins } from "./constants";
export var handleGridPadding = function handleGridPadding(noPadding) {
  if (noPadding) {
    return css(["padding:0;"]);
  } else {
    return css(["padding:0 ", "px;", "{padding:0 ", "px;}", "{padding:0 ", "px;}", "{padding:0 ", "px;}", "{padding:0 ", "px;}"], gridMargins.xs, media.sm, gridMargins.sm, media.md, gridMargins.md, media.lg, gridMargins.lg, media.xl, gridMargins.xl);
  }
};
var StyledGrid = styled.div.withConfig({
  displayName: "Grid__StyledGrid",
  componentId: "ovutvr-0"
})(["box-sizing:border-box;margin:0 auto;max-width:1472px;", " ", " ", ""], function (_ref) {
  var noPadding = _ref.noPadding;
  return handleGridPadding(!!noPadding);
}, function (_ref2) {
  var align = _ref2.align;

  if (align === 'left') {
    return "\n    margin-right: auto;\n    margin-left: 0;\n  ";
  }

  if (align === 'right') {
    return "\n    margin-left: auto;\n    margin-right: 0;\n  ";
  }
}, function (_ref3) {
  var fixedColumnWidth = _ref3.fixedColumnWidth;
  return fixedColumnWidth && css(["width:", "px;", "{width:", "px;}", "{width:", "px;}", "{width:", "px;}", "{width:", "px;}"], fixedColumnValues.xs.totalWidth, media.sm, fixedColumnValues.sm.totalWidth, media.md, fixedColumnValues.md.totalWidth, media.lg, fixedColumnValues.lg.totalWidth, media.xl, fixedColumnValues.xl.totalWidth);
});
export var Grid = function Grid(_ref4) {
  var _ref4$tag = _ref4.tag,
      tag = _ref4$tag === void 0 ? 'div' : _ref4$tag,
      rest = _objectWithoutProperties(_ref4, ["tag"]);

  return React.createElement(StyledGrid, _extends({
    as: tag
  }, rest));
};
//# sourceMappingURL=Grid.js.map