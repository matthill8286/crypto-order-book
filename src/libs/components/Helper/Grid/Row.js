function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import React from 'react';
import { media } from "../../../styles/media";
import { css, styled } from "../../../styles/styled";
import { gutterWidth } from "./constants";

var handleRowMarginBottom = function handleRowMarginBottom(noMargin) {
  if (noMargin) {
    return css(["margin-bottom:0;"]);
  } else {
    return css(["margin-bottom:", "px;", "{margin-bottom:", "px;}", "{margin-bottom:", "px;}", "{margin-bottom:", "px;}", "{margin-bottom:", "px;}"], gutterWidth.xs, media.sm, gutterWidth.sm, media.md, gutterWidth.md, media.lg, gutterWidth.lg, media.xl, gutterWidth.xl);
  }
};

var StyledRow = styled.div.withConfig({
  displayName: "Row__StyledRow",
  componentId: "sc-1o9ecac-0"
})(["display:flex;flex-flow:row wrap;align-items:stretch;margin:0;", " @supports (display:grid){display:grid;grid-gap:", "px;grid-template-columns:repeat(4,minmax(0,1fr));}", "{@supports (display:grid){grid-gap:", "px;grid-template-columns:repeat(8,minmax(0,1fr));}}", "{@supports (display:grid){grid-gap:", "px;}}", "{@supports (display:grid){grid-gap:", "px;grid-template-columns:repeat(12,minmax(0,1fr));}}", "{@supports (display:grid){grid-gap:", "px;}}"], function (_ref) {
  var noMargin = _ref.noMargin;
  return handleRowMarginBottom(!!noMargin);
}, gutterWidth.xs, media.sm, gutterWidth.sm, media.md, gutterWidth.md, media.lg, gutterWidth.lg, media.xl, gutterWidth.xl);
export var Row = function Row(_ref2) {
  var _ref2$tag = _ref2.tag,
      tag = _ref2$tag === void 0 ? 'div' : _ref2$tag,
      rest = _objectWithoutProperties(_ref2, ["tag"]);

  return React.createElement(StyledRow, _extends({
    as: tag
  }, rest));
};
//# sourceMappingURL=Row.js.map