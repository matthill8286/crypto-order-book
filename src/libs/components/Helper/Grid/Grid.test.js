import * as React from 'react';
import { renderWithTheme } from "../../../testRenderer";
import { Cell, Offset } from "./Cell";
import { Grid } from "./Grid";
import { Row } from "./Row";
describe('MaterialGrid', function () {
  var components = {
    Grid: Grid,
    Row: Row,
    Cell: Cell,
    Offset: Offset
  };
  describe('Grid component', function () {
    it('renders a div per default', function () {
      var tree = renderWithTheme(React.createElement(Grid, null));
      expect(tree).toMatchSnapshot();
    });
    it('renders with Rows and Cells (children)', function () {
      var tree = renderWithTheme(React.createElement(Grid, null, React.createElement(Row, null, React.createElement(Cell, null), React.createElement(Cell, null), React.createElement(Cell, null), React.createElement(Cell, null)), React.createElement(Row, null, React.createElement(Cell, null), React.createElement(Cell, null), React.createElement(Cell, null), React.createElement(Cell, null))));
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Cell component', function () {
    it('renders with custom responsive column count for view ports', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        colsXs: 4,
        colsSm: 8,
        colsMd: 8,
        colsLg: 12,
        colsXl: 12
      }));
      expect(tree).toMatchSnapshot();
    });
    it('renders with column order', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        order: 2
      }));
      expect(tree).toMatchSnapshot();
    });
    it('renders hidden with view port columns count set to 0', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        colsXs: 0,
        colsSm: 0,
        colsMd: 0,
        colsLg: 0,
        colsXl: 0
      }));
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Offset component', function () {
    it('renders with custom responsive column count for view ports', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        colsXs: 4,
        colsSm: 8,
        colsMd: 8,
        colsLg: 12,
        colsXl: 12
      }));
      expect(tree).toMatchSnapshot();
    });
    it('renders with column order', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        order: 2
      }));
      expect(tree).toMatchSnapshot();
    });
    it('renders hidden with view port columns count set to 0', function () {
      var tree = renderWithTheme(React.createElement(Cell, {
        colsXs: 0,
        colsSm: 0,
        colsMd: 0,
        colsLg: 0,
        colsXl: 0
      }));
      expect(tree).toMatchSnapshot();
    });
    it('renders hidden with no values, because of default props', function () {
      var tree = renderWithTheme(React.createElement(Offset, null));
      expect(tree).toMatchSnapshot();
    });
  });
  describe('additional customizations', function () {
    describe('with custom tags', function () {
      Object.keys(components).forEach(function (key) {
        it("renders ".concat(key, " with custom tag name"), function () {
          var Component = components[key];
          var tagName = 'section';
          var tree = renderWithTheme(React.createElement(Component, {
            tag: tagName
          }));
          expect(tree).toMatchSnapshot();
        });
      });
    });
  });
});
//# sourceMappingURL=Grid.test.js.map