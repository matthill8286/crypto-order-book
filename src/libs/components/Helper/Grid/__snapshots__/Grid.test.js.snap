// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialGrid Cell component renders hidden with view port columns count set to 0 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 kVHWOx"
/>
`;

exports[`MaterialGrid Cell component renders with column order 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 jixZHx"
  order={2}
/>
`;

exports[`MaterialGrid Cell component renders with custom responsive column count for view ports 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 cjVLej"
/>
`;

exports[`MaterialGrid Grid component renders a div per default 1`] = `
<div
  className="Grid__StyledGrid-ovutvr-0 cOfzdY"
/>
`;

exports[`MaterialGrid Grid component renders with Rows and Cells (children) 1`] = `
<div
  className="Grid__StyledGrid-ovutvr-0 cOfzdY"
>
  <div
    className="Row__StyledRow-sc-1o9ecac-0 leYzKt"
  >
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
  </div>
  <div
    className="Row__StyledRow-sc-1o9ecac-0 leYzKt"
  >
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
    <div
      className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
    />
  </div>
</div>
`;

exports[`MaterialGrid Offset component renders hidden with no values, because of default props 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 kVHWOx"
/>
`;

exports[`MaterialGrid Offset component renders hidden with view port columns count set to 0 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 kVHWOx"
/>
`;

exports[`MaterialGrid Offset component renders with column order 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 jixZHx"
  order={2}
/>
`;

exports[`MaterialGrid Offset component renders with custom responsive column count for view ports 1`] = `
<div
  className="Cellstyled__StyledCell-sc-4z19ev-0 cjVLej"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Cell with custom tag name 1`] = `
<section
  className="Cellstyled__StyledCell-sc-4z19ev-0 jNLgpA"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Grid with custom tag name 1`] = `
<section
  className="Grid__StyledGrid-ovutvr-0 cOfzdY"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Offset with custom tag name 1`] = `
<section
  className="Cellstyled__StyledCell-sc-4z19ev-0 kVHWOx"
/>
`;

exports[`MaterialGrid additional customizations with custom tags renders Row with custom tag name 1`] = `
<section
  className="Row__StyledRow-sc-1o9ecac-0 leYzKt"
/>
`;
